pkg/lib/cockpit.js

```diff
@@ -3700,6 +3721,41 @@
         return self;
     };
 
+    cockpit.fsinfo = function (path, attrs, options = {}) {
+        const self = {
+            value: null
+        };
+
+        const callbacks = [];
+
+        function on_message(_event, payload) {
+            const msg = JSON.parse(payload);
+            self.value = json_merge(self.value, msg);
+            for (const callback of callbacks) {
+                callback(self.value);
+            }
+        }
+
+        self.effect = function(callback) {
+            callback(self.value);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19845-20240115-145734-3653cecb-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)